src/main/java/com/linecorp/armeria/server/metrics/ServiceMetricConsumer.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2015 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.metrics;
+
+import java.util.Objects;
+
+@FunctionalInterface
+public interface ServiceMetricConsumer {
+    /**
+     * Invoked for each request has been processed
+     *
+     * @param staus flag whether it's successful or not
+     * @param processNanoTime elapsed nano time processing request
+     * @param requestSize number of bytes in request if possible, otherwise it will be 0
+     * @param responseSize number of bytes in response if possible, otherwise it will be 0
+     */
+    void onComplete(Status staus, long processNanoTime, int requestSize, int responseSize);
```

Comment:
I wonder if it's a good idea to have more detailed information about the status of the invocation. Perhaps we could make `Status` extensible? e.g.

```java
public class Status {
    ... some enum-ish operations ...
    public SessionProtocol sessionProtocol() {
         ...
    }
}

public class HttpStatus extends Status {
    ...
    public static HttpStatus of(int code) {
        ...
    }

    public int code() {
        ...
    }
}

ServiceMetricConsumer c = (status, ....) -> {
    if (SessionProtocol.ofHttp().contains(status.sessionProtocol())) {
        HttpStatus hs = (HttpStatus) status;
        ...
    } else {
        ...
    }
};
```

Alternatively, we could refrain from introducing a subtype like `HttpStatus` and just add a method that returns the suffix of the metric registry key:

```java
public final class Status {
    public static of(SessionProtocol sessProto, int code) { ... } // Lazy caching

    public String text() {
        return sessionProtocol.uriText() + '.' + code;
    }
}
```